com_redhat_docker/ks/docker.py

```diff
@@ -43,6 +43,123 @@
 
 # pylint: disable=interruptible-system-call
 
+class LVMStorage(object):
+    def __init__(self, addon):
+        self.addon = addon
+
+    @property
+    def dm_fs(self):
+        """ Return the docker dm.fs value """
+        if not self.addon:
+            return ""
+        return "dm.fs=%s" % self.addon.fstype
+
+    @property
+    def pool_name(self):
+        """ Return the docker dm.thinpool value """
+        if not self.addon:
+            return ""
+        return "dm.thinpooldev=/dev/mapper/%s-docker--pool" % self.addon.vgname
+
+    @property
+    def addon_str(self):
+        """ Return the addon string and storage driver options """
+        return '%%addon %s --vgname="%s" --fstype="%s"' % (self.addon.name, self.addon.vgname, self.addon.fstype)
+
+    def check_setup(self, storage, ksdata, instClass):
+        """ Check storage to make sure the selected VG has a thinpool and it is named docker-pool
+
+        :param storage: Blivet storage object
+        :param ksdata: Kickstart data object
+        :param instClass: Anaconda installclass object
+
+        If there is an error, raise the appropriate Kickstart error.
+        """
+        if self.addon.vgname not in (vg.name for vg in storage.vgs):
+            raise KickstartValueError(formatErrorMsg(0, msg=_("%%addon com_redhat_docker is missing VG named %s")) % self.addon.vgname)
+
+        # Make sure the VG has a docker-pool LV
+        if self.addon.vgname+"-docker-pool" not in (lv.name for lv in storage.lvs):
+            raise KickstartValueError(formatErrorMsg(0, msg=_("%%addon com_redhat_docker is missing a LV named docker-pool")))
```

Comment:
Right, but it will still be available to be used. I'll probably change everything in lmc and this when the new pyks is available.